### ISSUE TITLE
Train RNet error

### ISSUE THREAD

**Issue by MarekPetr** [look4pritam/tfmtcnn#4]
Cannot feed value of shape (320, 24, 24, 3) for Tensor 'input_image:0', which has shape '(384, 24, 24, 3).

**Comment by look4pritam** [comment]
384 is batch size for training. You may be having only 320 samples for
training.

**Comment by MarekPetr** [comment]
Thanks for a fast answer, but that does not solve the problem. Problem is, that the batch feed value changes with every sample. I even tried to change it down to one, to cover them all, but it produces another error. I have read, that it is possilble to implement variable batch size, which I think is exactly what it wants, but I did not figure out how to implement it in your code.

**Comment by look4pritam** [comment]
Check for CelebA branch, it contains latest code. Here you will not face this problem.
I have optimized this branch till PNet (95% accuracy on FDDB dataset) whereas RNet and ONet are under process.

**Comment by MarekPetr** [comment]
Thank you, I had a look at that branch and it has exactly the same code for the train method. But I will try it after I change every line of code to get rid of combining spaces and tabs again.

**Comment by look4pritam** [comment]
Training code is same but data generation code is different. So that batch size is always 384 and it has sufficient data.

**Comment by look4pritam** [comment]
I have not implemented variable size batch.

**Comment by MarekPetr** [comment]
What do you mean with data generation? Datasets generated for PNet, RNet and ONet? I just wonder what would I have to do again to make it work with my already trained PNet and generated dataset for RNet.

**Comment by look4pritam** [comment]
Dataset for RNet is generated using PNet model (hard image samples) + random image samples.

**Comment by look4pritam** [comment]
---But I will try it after I change every line of code to get rid of combining spaces and tabs again.
Is there any automatic way?
I am trying to impose some coding standard for python.

**Comment by MarekPetr** [comment]
I thought that RNet is generated using PNet and random samples in master brach as well. And the spaces and tabs could be easily solved if there was the same amount of spaces. For example if you indent with 4 spaces, you could just replace every 4 spaces in a row. But thats not the case, so it is 
 often manual work. Python needs either spaces or tabs to indent, but it does not allow to combine them. We could chat on another place and just paste here the results we came to, if you dont mind.

**Comment by MarekPetr** [comment]
Thank you so much, I will try it right after exams :)
